### PR TITLE
Update issue templates

### PR DESCRIPTION
**Pull request by elliedavidson**
This PR: 
* Adds a general issue template to help us better track our work

Once this is merge I will create an example issue and link it to the issue template.

### CODE REVIEW

**Comment by jbearer** [comment]
I like it. Should we add an `estimate` field?

**Comment by elliedavidson** [comment]
> I like it. Should we add an `estimate` field?

Yes, I like that idea!

**Comment by ggutoski** [comment]
> 10 pending reviewers
😆 Not part of this PR but perhaps we should leave it to the author to select 1-2 reviewers instead of spamming all code owners.

File: .github/ISSUE_TEMPLATE/hotshot-general-task-template.md
Comment (shenkeyao) [inline .github/ISSUE_TEMPLATE/hotshot-general-task-template.md:10]: We could also add these to the description:
* Task dependencies
* Solution trade-offs

File: .github/ISSUE_TEMPLATE/hotshot-general-task-template.md
Comment (shenkeyao) [inline .github/ISSUE_TEMPLATE/hotshot-general-task-template.md:24]: Maybe also add `[MACRO_TASK_SHORT_NAME]` in the description, so that its subtasks have a consistent short name to use in their names.

File: .github/ISSUE_TEMPLATE/hotshot-general-task-template.md
Comment (shenkeyao) [inline .github/ISSUE_TEMPLATE/hotshot-general-task-template.md:25]: How do we feel about a nested subtasks? I don't have a strong opinion here, but it may be a good idea to avoid too many layers of tasks, so if we have something like:
> Macro task A -> subtask B -> subtasks C and D

We could change it to either
> Macro task A -> subtask B -> _bullet points_ C and D

or
> Macro task A -> subtask B, _C, and D_

for easier tracking.

**Review by shenkeyao — COMMENTED** [review]
Looks good overall! Left some additional thoughts.

**Comment by elliedavidson** [comment]
> > 10 pending reviewers
> 
> 😆 Not part of this PR but perhaps we should leave it to the author to select 1-2 reviewers instead of spamming all code owners.

Point taken 😄

File: .github/ISSUE_TEMPLATE/hotshot-general-task-template.md
Comment (elliedavidson) [reply .github/ISSUE_TEMPLATE/hotshot-general-task-template.md:25]: Yes, I like all of these suggestions!

**Comment by ggutoski** [comment]
> Point taken 😄

oh snap I thought all those review requests were automatically tagged by github. oops

**Review by DieracDelta — APPROVED** [review]
This is really great! +1 +1

**Comment by elliedavidson** [comment]
Closing in favor of another PR.

**Pull request closed without merging**
